### PR TITLE
Report jest test results to AzDo

### PR DESCRIPTION
**Pull request by dibarbet**
Looks like

image

### CODE REVIEW

File: test/runIntegrationTests.ts [hunk 1]
```diff
@@ -24,19 +24,18 @@ export async function runIntegrationTests(projectName: string) {
         verbose: true,
     } as Config.Argv;
 
-    let filter: string;
     if (process.env.TEST_FILE_FILTER) {
-        // If we have just a file, run that with runTestsByPath.
-        jestConfig.runTestsByPath = true;
+        // If we have just a file, run that with an explicit match.
```
Comment (dibarbet) [inline test/runIntegrationTests.ts:28]: None of the changes here are explicitly related to the test reporting.  But while trying to get that to work I fixed the issue where the launch.json threw out a ton of errors even when the tests passed.

Now looks like
image

File: jest.config.ts [hunk 2]
```diff
@@ -14,6 +14,23 @@ const config: Config = {
...
+    reporters: [
+        'default',
+        [
+            'jest-junit',
+            {
+                outputDirectory: '<rootDir>/out/',
+                reportTestSuiteErrors: 'true',
+                // Azure DevOps does not display test suites (it ignores them entirely).
+                // So we have to put all the info in the test case name so the UI shows anything relatively useful.
+                // See https://github.com/microsoft/azure-pipelines-tasks/issues/7659
```
Comment (dibarbet) [inline jest.config.ts:29]: 😢

File: .vscode/launch.json [hunk 3]
```diff
@@ -41,7 +41,8 @@
                 "${workspaceFolder}/**",
                 "!**/node_modules/**"
             ],
-            "preLaunchTask": "buildDev"
+            "preLaunchTask": "buildDev",
+            "internalConsoleOptions": "openOnSessionStart"
```
Comment (dibarbet) [inline .vscode/launch.json:45]: Not related to the reporting - just an improvement to open the debug output (where test results are printed) when running the integration tests

**Comment by dibarbet** [comment]
There is an unfortunate issue with jest where if a afterall/beforeall causes a failure, the failure cannot be linked to a specific test.  This appears to be the case for the Razor formatting test.  This looks like
image

**Review by jasonmalinowski — APPROVED** [review]

**Comment by davidwengier** [comment]
> There is an unfortunate issue with jest where if a afterall/beforeall causes a failure, the failure cannot be linked to a specific test. This appears to be the case for the Razor formatting test.

Seems like making the display clearly different is a good thing, since if its in afterall/beforeall that means its not my fault, right? 😛

**Pull request merged**